operator/builtin/input/file/reader.go

```diff
@@ -150,10 +151,37 @@ func (f *Reader) emit(ctx context.Context, msgBuf []byte) error {
 	if err := e.Set(f.fileInput.FileNameField, filepath.Base(f.Path)); err != nil {
 		return err
 	}
+	if err := f.resolvePath(e); err != nil {
+		return err
+	}
+
 	f.fileInput.Write(ctx, e)
 	return nil
 }
 
+// resolvePath resolves path to real and absolute form and add them into entry
+func (f *Reader) resolvePath(e *entry.Entry) error {
+	resolved, err := filepath.EvalSymlinks(f.Path)
+	if err != nil {
+		return err
+	}
+
+	abs, err := filepath.Abs(resolved)
```

Comment:
Do you expect a performance hit here?

This could be computed once per Reader, but that would mean it does not detect changes in symlinks. So perhaps the next best thing would be to check if it is needed at all. Since the default configuration does not need it, this would potentially avoid a performance hit for many users.